includes/Core/Modules/Module.php

```diff
@@ -319,15 +320,16 @@ public function is_connected() {
 				continue;
 			}
 
-			if ( empty( $datapoint_services[ $datapoint ] ) ) {
+			$datapoint_service = $datapoint_definitions[ $definition_key ]['service'];
+			if ( empty( $datapoint_service ) ) {
 				continue;
 			}
 
-			if ( ! isset( $service_batches[ $datapoint_services[ $datapoint ] ] ) ) {
-				$service_batches[ $datapoint_services[ $datapoint ] ] = $this->google_services[ $datapoint_services[ $datapoint ] ]->createBatch();
+			if ( ! isset( $service_batches[ $datapoint_service ] ) ) {
+				$service_batches[ $datapoint_service ] = $this->google_services[ $datapoint_service ]->createBatch();
 			}
 
-			$service_batches[ $datapoint_services[ $datapoint ] ]->add( $request, $key );
+			$service_batches[ $datapoint_service ]->add( $request, $key );
 			$results[ $key ] = $datapoint;
```

Comment:
Since we're now differentiating between the request method (see above `{$dataset->method}:{$dataset->datapoint}`), I think we should do that here as well, i.e. store both `$dataset->method` and `$datapoint` here so that we can reference that in the loop below for handling the exception.

Since the value here is only temporarily set and will always be overridden, changing the format doesn't matter in terms of backward compatibilty.

Reply:
Yep, agree, updated.